### DISCUSSION TITLE
Rule engine data transformation (date to epoch )

### DISCUSSION THREAD

**Discussion by and-dzh3** [emqx/emqx#5748] [Q&A]
Hi how i can use rule engine to  transform   timestamp format from :"2021-08-26 19:05:18.327" to unix time stamp. i gona use republish option

**Comment by k32** [comment]
Hello, 
Have you tried
```
SELECT rfc3339_to_unix_ts(payload.timestamp) as timestamp
FROM ...
```
?

**Reply by zmstone** [reply]
functions are defined here:
https://github.com/emqx/emqx-rule-engine/blob/master/src/emqx_rule_funcs.erl

**Reply by zmstone** [reply]
sorry, I provided the link to a wrong repo, the latest should be here: https://github.com/emqx/emqx/blob/master/apps/emqx_rule_engine/src/emqx_rule_funcs.erl

**Reply by and-dzh3** [reply]
Bad Arguments: {select_and_transform_error,{{sql_function_not_supported,<<"rfc3339_to_unix_ts">>},[{emqx_rule_runtime,apply_func
emqx version 4.2.13

**Comment by and-dzh3** [comment]
Thanks I will try it .Also I need to do some modifications to one of the field (remove 4 chars) … all of this thing are not documented and there is no example of how to do it ….

**Reply by zmstone** [reply]
hi. all functions are documented here: https://docs.emqx.io/en/enterprise/v4.3/rule/rule-engine.html#sql-statement

**Comment by zmstone** [comment]
we'll continue to refine the documents, to provide more examples,
and also to make the rule engine more extendable.

the `rfc3339_to_unix_ts` is unfortunately missing from the document.
to remove 4 chars, you can use the `substr` function (depending on where these chars are).

**Reply by and-dzh3** [reply]
Bad Arguments: {select_and_transform_error,{{sql_function_not_supported,<<"rfc3339_to_unix_ts">>},[{emqx_rule_runtime,apply_func
emqx version 4.2.13

**Reply by k32** [reply]
This function was added in 4.3

**Reply by and-dzh3** [reply]
in 4.3  , error 500

**Reply by k32** [reply]
Could you show the broker logs? I tested the SQL query from the first answer at 4.3.8
